### PR TITLE
Raise an exception if no SQLAlchemy session was set

### PR DESCRIPTION
**Pull request by llazzaro**
With unit tests

### CODE REVIEW

File: factory/alchemy.py [hunk 1]
```diff
@@ -91,6 +91,8 @@ def _create(cls, model_class, *args, **kwargs):
             session_persistence = SESSION_PERSISTENCE_FLUSH
 
         obj = model_class(*args, **kwargs)
+        if session is None:
+            raise RuntimeError('No session provided.')
```
Comment (jeffwidman) [inline factory/alchemy.py]: Nit: Do you mind using double quotes for this message? 

Both myself and @rbarrois prefer double quotes for strings intended for human consumption, and single-quotes for machine consumption.

Comment (jeffwidman) [inline factory/alchemy.py]: Nit: Do you mind using double quotes for this message?

Both myself and @rbarrois prefer double quotes for strings intended for human consumption, and single-quotes for machine consumption.

**Comment by jeffwidman** [comment]
Looks perfect other than the minor quotes nit. 

Thanks for putting this together.

File: factory/alchemy.py (hunk 1 above)
Comment (llazzaro) [inline factory/alchemy.py]: sure!

**Comment by llazzaro** [comment]
@jeffwidman added double quotes.

**Comment by jeffwidman** [comment]
Thanks again! 

BTW, on github no need to create a new commit... just `git commit --amend` to pickup your staged change, then `git push -f` to force push to the branch (since you're effectively replacing one commit with another) and it will show up immediately here in the pull request.

**Pull request merged**